glean-core/src/metrics/jwe.rs

```diff
@@ -0,0 +1,374 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+use std::fmt;
+use std::str::FromStr;
+
+use serde::Serialize;
+
+use crate::error_recording::{record_error, ErrorType};
+use crate::metrics::{Metric, MetricType};
+use crate::storage::StorageManager;
+use crate::CommonMetricData;
+use crate::Glean;
+
+const DEFAULT_MAX_CHARS_PER_ELEMENT: usize = 1024;
+
+/// Verifies if a string is [`BASE64URL`](https://tools.ietf.org/html/rfc4648#section-5) compliant.
+///
+/// As such, the string must match the regex: `[a-zA-Z0-9\-\_]*`.
+///
+/// > **Note** As described in the [JWS specification](https://tools.ietf.org/html/rfc7516#section-2),
```

Comment:
Is this URL correct? Where is the discarding part specified?

Reply:
https://tools.ietf.org/html/rfc7515#section-2, off by one. Nice catch!

Reply:
Mh, by looking at the referenced [sample implementation](https://tools.ietf.org/html/rfc7515#appendix-C), looks like `=` is still an allowed character.

Reply:
It is allowed, BASE64URL may have the `=`. For the purposes of JWE and JWS though, the padding is not present. From the RFC:
> Base64 encoding using the URL- and filename-safe character set defined in Section 5 of RFC 4648 [RFC4648], **with all trailing '=' characters omitted (as permitted by Section 3.2)** [...]

Notice that the sample implementation removes the trailing `=` upon encoding, and re-adds them upon decoding. 

Since we need to validate that a string is valid BASE64URL (without padding) and not encode or decode, we don't expect any `=`.